romancal/tweakreg/tests/test_tweakreg.py

```diff
@@ -1232,18 +1234,80 @@ def test_update_source_catalog_coordinates(tmp_path, base_image):
         cat.source_catalog["xcentroid"], cat.source_catalog["ycentroid"]
     )
-    expected_psf = img.meta.wcs.pixel_to_world(
+    expected_psf = img.meta.wcs(
         cat.source_catalog["x_psf"], cat.source_catalog["y_psf"]
     )
 
     # compare coordinates (make sure tweaked WCS was applied to cat file coords)
-    np.testing.assert_array_equal(cat_ra_centroid, expected_centroid.ra.value)
-    np.testing.assert_array_equal(cat_dec_centroid, expected_centroid.dec.value)
-    np.testing.assert_array_equal(cat_ra_psf, expected_psf.ra.value)
-    np.testing.assert_array_equal(cat_dec_psf, expected_psf.dec.value)
+    np.testing.assert_array_equal(cat_ra_centroid, expected_centroid[0])
+    np.testing.assert_array_equal(cat_dec_centroid, expected_centroid[1])
+    np.testing.assert_array_equal(cat_ra_psf, expected_psf[0])
+    np.testing.assert_array_equal(cat_dec_psf, expected_psf[1])
+
+
+def test_source_catalog_coordinates_have_changed(tmp_path, base_image):
+    """Test that the original catalog file content is different from the updated file."""
+
+    os.chdir(tmp_path)
+
+    img = base_image(shift_1=1000, shift_2=1000)
+    add_tweakreg_catalog_attribute(tmp_path, img, catalog_filename="img_1")
+
+    tweakreg = trs.TweakRegStep()
+
+    # create SourceCatalogModel
+    source_catalog_model = setup_source_catalog_model(img)
+
+    # save SourceCatalogModel
+    tweakreg.save_model(
+        source_catalog_model,
+        output_file="img_1.asdf",
+        suffix="cat",
+        force=True,
+    )
+    # save original data
+    shutil.copy("img_1_cat.asdf", "img_1_cat_original.asdf")
+
+    # update tweakreg catalog name
+    img.meta.source_detection.tweakreg_catalog_name = "img_1_cat.asdf"
+
+    # run TweakRegStep
+    res = trs.TweakRegStep.call([img])
+
+    # tweak the current WCS using TweakRegStep and save the updated cat file
+    with res:
+        dm = res.borrow(0)
+        assert dm.meta.source_detection.tweakreg_catalog_name == "img_1_cat.asdf"
+        tweakreg.update_catalog_coordinates(
+            dm.meta.source_detection.tweakreg_catalog_name, dm.meta.wcs
+        )
+        res.shelve(dm, 0)
+
+    cat_original = rdm.open("img_1_cat_original.asdf")
+    cat_updated = rdm.open("img_1_cat.asdf")
+
+    # compare contents
+    assert not filecmp.cmp("img_1_cat.asdf", "img_1_cat_original.asdf")
+    # compare coordinates using rtol=1e-07 and atol=0 (default)
+    np.testing.assert_allclose(
+        cat_original.source_catalog["ra_centroid"],
+        cat_updated.source_catalog["ra_centroid"],
+    )
+    np.testing.assert_allclose(
+        cat_original.source_catalog["dec_centroid"],
+        cat_updated.source_catalog["dec_centroid"],
+    )
+    np.testing.assert_allclose(
+        cat_original.source_catalog["ra_psf"],
+        cat_updated.source_catalog["ra_psf"],
+    )
+    np.testing.assert_allclose(
+        cat_original.source_catalog["dec_psf"],
+        cat_updated.source_catalog["dec_psf"],
+    )
```

Comment:
Can you help me with this test?  The test is that the files are not bitwise identical, but the ras and decs didn't change meaningfully?  I was kind of expecting this test to show that the ras and decs _do_ change, and I read this test as showing the opposite; am I missing something?

I don't know what precision works, but I'd drop the filecmp.cmp test, and maybe make the tests look like:
```
for column in [ra, dec, centroid, psf]:
    assert np.allclose(orig.catalog[column], new.catalog[column], atol=1/60/60)  # assert nothing moved by more than an arcsecond
    assert not np.allclose(orig.catalog[column], new.catalog[column], atol=1/60/60/100)  # assert some things moved by at least a hundredth of an arcsecond
```
I'm kind of guessing what the right precisions are---it depends on how many stars are in the catalog, and how much you shifted their positions.  But we want to be showing that we did tweak the positions, and I feel like the current test is kind of showing the opposite?

Note 1e-7 rtol corresponds to roughly 0.1" or 1 pixel.  Do you know how big of a tweak you're applying?

Reply:
The lines above check that the tweaked coordinates are not too far off (they should be within 1 pixel, since the set up of the mock catalog adds random shifts between -0.5 and 0.5 pixel). We have 22 sources in the catalog. I set `atol` and `rtol` to correspond to roughly half a pixel.

The test to prove that things did move was indeed missing and has been added in the updates to this PR. We are checking if things have moved by at least 1/100th of a pixel.